omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,300 @@
+                errors.add("base character set"); 
+            }
+            if (description != null && StringUtils.isNotBlank(description.toString())) {
+                identifierSource.setDescription(description.toString());
+            }
+            if (prefix != null && StringUtils.isNotBlank(prefix.toString())) { 
+                identifierSource.setPrefix(prefix.toString()); 
+            }
+            if (suffix != null && StringUtils.isNotBlank(suffix.toString())) { 
+                identifierSource.setSuffix(suffix.toString()); 
+            }
+            if (minLength != null && StringUtils.isNotBlank(minLength.toString())) {
+                if(StringUtils.isNumeric(minLength.toString())) {
+                        identifierSource.setMinLength(Integer.parseInt(minLength.toString()));
+                }
+            }
+            if (maxLength != null && StringUtils.isNotBlank(maxLength.toString())) {
+                if(StringUtils.isNumeric(maxLength.toString())) {
+                        identifierSource.setMaxLength(Integer.parseInt(maxLength.toString()));
+                }
+            }
+            if(errors.toString().replaceAll("[\\[\\]]", "").length() > 0) {
+                throw new ValidationException("The values of the following inputs are missing or invalid: " + errors.toString());
+            }
+
+            identifierSource.setIdentifierType(patientIdentifierType);
+            identifierSource.setName(name.toString());
+            identifierSource.setBaseCharacterSet(baseCharacterSet.toString());
+            identifierSource.setFirstIdentifierBase(firstIdentifierBase.toString());
+            savedIdentifierSource =  save(identifierSource);
+        }
+        else if(identifierSourceType.equals("RemoteIdentifierSource")) {
+            RemoteIdentifierSource identifierSource = new RemoteIdentifierSource();
+            Object username = postBody.get("username");
+            Object password = postBody.get("password");
+            Object url = postBody.get("url");
+            
+            if (url == null || StringUtils.isBlank(url.toString())) { 
+                    errors.add("url"); 
+            }
+            if (description != null && StringUtils.isNotBlank(description.toString())) {
+                identifierSource.setDescription(description.toString());
+            }
+            if (username!= null && StringUtils.isNotBlank(username.toString())) { 
+                    identifierSource.setUser(username.toString()); 
+            }
+            if (password != null && StringUtils.isNotBlank(password.toString())) {
+                if(username == null || StringUtils.isBlank(username.toString())) {
+                    errors.add("username");
+                }
+                else {
+                    identifierSource.setPassword(password.toString());
+                }
+            }
+            if(errors.toString().replaceAll("[\\[\\]]", "").length() > 0) {
```

Comment:
Am struggling to understand why you need to first call toString() is all you want to know is how many items are in this list

Reply:
replaced with: `if(errors.size() > 0) {}`

My mind was stuck on that one: 😊